### PR TITLE
[Docs] Document IP filtering for RCS 2.0

### PR DESCRIPTION
**Pull request by ywangd**
This PR adds documentation for RCS 2.0 IP filtering.

Relates: #98472

### CODE REVIEW

File: docs/reference/settings/security-settings.asciidoc [hunk 1]
```diff
@@ -2609,7 +2609,7 @@ setting, this would be `transport.profiles.$PROFILE.xpack.security.ssl.key`.
 
 [discrete]
 [[ip-filtering-settings]]
-==== IP filtering settings
+=== IP filtering settings
```
Comment (ywangd) [inline docs/reference/settings/security-settings.asciidoc:2612]: I think the existing indentation level is incorrect for the right sidebar. This section should be at the same level as TLS settings.

File: docs/reference/settings/security-hash-settings.asciidoc [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [discrete]
 [[hashing-settings]]
-==== User cache and password hash algorithms
+=== User cache and password hash algorithms
```
Comment (ywangd) [inline docs/reference/settings/security-hash-settings.asciidoc:3]: Same here as well.

File: x-pack/docs/en/security/using-ip-filtering.asciidoc [hunk 3]
```diff
@@ -110,9 +111,32 @@ xpack.security.http.filter.allow: 172.16.0.0/16
...
+
+NOTE: Whether IP filtering for remote cluster is enabled is controlled by
+`xpack.security.transport.filter.enabled` as well. This means filtering for
+the remote cluster and transport interfaces must be enabled or disabled together.
+But the exact allow and deny lists can be different between them.
+
 [discrete]
 [[dynamic-ip-filtering]]
-==== Dynamically updating IP filter settings
+=== Dynamically updating IP filter settings
```
Comment (ywangd) [inline x-pack/docs/en/security/using-ip-filtering.asciidoc:139]: Similarly here as well.

**Review by n1v0lg — APPROVED** [review]
LGTM. Build failures all look related to version bump fallout from a few days.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@abdonpijpelink Could you please take a look at this PR when convenient? Thanks!

**Review by abdonpijpelink — APPROVED** [review]
LGTM. I've left two suggestions.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Pull request merged**